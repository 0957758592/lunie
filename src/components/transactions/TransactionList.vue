<template>
  <div>
    <TransactionItem
      v-for="msg in transactions"
      :key="msg.key"
      :transaction="msg"
      :validators="validators"
      :address="address"
    />
  </div>
</template>

<script>
// import TransactionItem from "./TransactionItem"
import TransactionItem from "./TransactionItem"
import { getUnbondTimeFromTX } from "scripts/time"

export default {
  name: `transaction-list`,
  components: {
    TransactionItem
  },
  props: {
    validatorsUrl: {
      type: String,
      default: null
    },
    proposalsUrl: {
      type: String,
      default: null
    },
    transactions: {
      type: Array,
      required: true
    },
    address: {
      type: String,
      default: null
    },
    bondingDenom: {
      type: String,
      required: true
    },
    validators: {
      type: Object,
      required: true
    },
    unbondingDelegations: {
      type: Object,
      required: true
    }
  },
  methods: {
    getUnbondTimeFromTX,
    isBankTx(type) {
      return [`cosmos-sdk/MsgSend`].includes(type)
    },
    isStakingTx(type) {
      return [
        `cosmos-sdk/MsgCreateValidator`,
        `cosmos-sdk/MsgEditValidator`,
        `cosmos-sdk/MsgDelegate`,
        `cosmos-sdk/MsgUndelegate`,
        `cosmos-sdk/MsgBeginRedelegate`,
        `cosmos-sdk/MsgUnjail`
      ].includes(type)
    },
    isGovernanceTx(type) {
      return [
        `cosmos-sdk/MsgSubmitProposal`,
        `cosmos-sdk/MsgDeposit`,
        `cosmos-sdk/MsgVote`
      ].includes(type)
    },
    isDistributionTx(type) {
      return [
        `cosmos-sdk/MsgSetWithdrawAddress`,
        `cosmos-sdk/MsgWithdrawDelegationReward`,
        `cosmos-sdk/MsgWithdrawValidatorCommission`
      ].includes(type)
    }
  }
}
</script>
