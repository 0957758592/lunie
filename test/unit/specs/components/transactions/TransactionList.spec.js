import { shallowMount } from "@vue/test-utils"
import TransactionList from "src/components/transactions/TransactionList"

describe(`TransactionList`, () => {
  let wrapper

  const transactions = [
    {
      blockNumber: 1086769,
      fees: {
        amount: "37",
        denom: "uatom"
      },
      group: "banking",
      key: "keyhash",
      memo: "(Sent via Lunie)",
      time: "Thu Jul 18 2019 12:03:11 GMT+0200 (Central European Summer Time)",
      liquidDate: NaN,
      type: "cosmos-sdk/MsgSend"
    }
  ]

  beforeEach(() => {
    wrapper = shallowMount(TransactionList, {
      propsData: {
        transactions: transactions,
        validators: {},
        address: "cosmos1"
      }
    })
  })

  it(`has the expected html structure`, () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
