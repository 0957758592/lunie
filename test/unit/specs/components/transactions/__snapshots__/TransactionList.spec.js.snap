// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionList has the expected html structure 1`] = `
<div>
  <transactionitem-stub
    address="cosmos1"
    transaction="[object Object]"
    validators="[object Object]"
  />
</div>
`;
